syncer/syncer.go

```diff
@@ -395,25 +395,22 @@ func (s *Syncer) Init() (err error) {
 	}
 	rollbackHolder.Add(fr.FuncRollback{Name: "remove-active-realylog", Fn: s.removeActiveRelayLog})
 
-	// init successfully, close done chan to make Syncer can be closed
-	// when Process started, we will re-create done chan again
-	// NOTE: we should refactor the Concurrency Model some day
-	s.done = make(chan struct{})
-	close(s.done)
+	s.done = nil
```

Comment:
we can set `s.done = nil` in `NewSyncer`?

Reply:
if set `s.done = nil` in `NewSyncer` , syncer can close before some init function, but old logic is wait until some init succeed, I'm not sure is this OK?

Reply:
what does `old logic` stand for?